src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java

```diff
@@ -61,4 +119,138 @@ public IssuerDTO getIssuerConfig(String issuerId) throws ApiNotAccessibleExcepti
+        credentialSubject.keySet().forEach(VCProperty -> vcPropertiesFromWellKnown.put(VCProperty, credentialSubject.get(VCProperty).getDisplay().get(0).getName()));
+        String backgroundColor = credentialsSupportedResponse.getDisplay().get(0).getBackground_color();
+        String textColor = credentialsSupportedResponse.getDisplay().get(0).getText_color();
+        VCCredentialRequest vcCredentialRequest = generateVCCredentialRequest(issuerDTO, credentialsSupportedResponse, accessToken);
+        logger.info("VC Credential Request is -> " + vcCredentialRequest);
+        //Esignet API call for credential issue
+        VCCredentialResponse vcCredentialResponse = restApiClient.postApi(credentialEndPoint, MediaType.APPLICATION_JSON,
+                vcCredentialRequest, VCCredentialResponse.class, accessToken);
+        logger.info("VC Credential Response is -> " + vcCredentialResponse);
+        if (vcCredentialResponse == null) throw new RuntimeException("VC Credential Issue API not accessible");
+        Map<String, Object> credentialProperties = vcCredentialResponse.getCredential().getCredentialSubject();
+        LinkedHashMap<String,Object> displayProperties = new LinkedHashMap<>();
+        vcPropertiesFromWellKnown.keySet().forEach(vcProperty -> displayProperties.put(vcPropertiesFromWellKnown.get(vcProperty), credentialProperties.get(vcProperty)));
+        return getPdfResourceFromVcProperties(displayProperties, textColor, backgroundColor,
+                credentialsSupportedResponse.getDisplay().get(0).getName(),
+                issuerDTO.getDisplay().stream().map(d -> d.getLogo().getUrl()).findFirst().orElse(""));
+    }
+
+    private VCCredentialRequest generateVCCredentialRequest(IssuerDTO issuerDTO, CredentialsSupportedResponse credentialsSupportedResponse, String accessToken) throws ParseException {
+        //Getting public key from the p12 file
+        PublicKey publicKeyString = joseUtil.getPublicKeyString(keyStorePath, fileName, issuerDTO.getClient_alias(), cyptoPassword);
+        //Generating proof from the public key with custom header
+        String jwt = joseUtil.generateJwt(publicKeyString, keyStorePath, fileName, issuerDTO.getClient_alias(),
+                cyptoPassword, issuerDTO.getCredential_audience(), issuerDTO.getClient_id(), accessToken);
+        return VCCredentialRequest.builder()
+                .format(credentialsSupportedResponse.getFormat())
+                .proof(VCCredentialRequestProof.builder()
+                        .proof_type(credentialsSupportedResponse.getProof_types_supported().get(0))
+                        .jwt(jwt)
+                        .build())
+                .credential_definition(VCCredentialDefinition.builder()
+                        .type(credentialsSupportedResponse.getCredential_definition().getType())
+                        .context(List.of("https://www.w3.org/2018/credentials/v1"))
+                        .build())
```

Comment:
Extract this as constant

Reply:
made a static variable and referenced here